debezium-core/src/main/java/io/debezium/schema/SchemaChangeEvent.java

```diff
@@ -37,13 +38,13 @@ public class SchemaChangeEvent {
     private TableChanges tableChanges = new TableChanges();
 
     public SchemaChangeEvent(Map<String, ?> partition, Map<String, ?> offset, Struct source, String database, String schema, String ddl, Table table,
-                             SchemaChangeEventType type, boolean isFromSnapshot) {
+                             SchemaChangeEventType type, boolean isFromSnapshot, TableId previousTableId) {
```

Comment:
@Naros Is it possible to rename table and at the same time change its structure? If not then we should probably handle ranme as a distinct operation and either introduce `RENAME` change type or instead of having constructors we should provide a set of factory methods.

The code as is could probably go into 1.9 but as we have more flexibility for 2.0 we should re-consider this from architectural PoV.

Reply:
So I know specifically with Oracle there are 2 ways to rename a table, the legacy way is a `RENAME ...` statement and the other is to use an `ALTER TABLE RENAME ...` statement.  I've never seen any documentation that indicates that you can adjust the table's structure while also renaming it with Oracle.  In fact, I just looked at the DDL syntax generated in the docs and it seems to indicate they're mutually exclusive.

I think it would be nice to have a `RENAME` type, but I'm concerned about backward compatibility.  I think we wanted to tackle this change so that if a user upgraded and then downgraded the records would still be consumed without any problems.

Reply:
@Naros Hence what I am saying - this is fine for 1.9 - keep compatibility and make it better for 2.0 even at cost of compatibility break.